local_planner/src/nodes/local_planner.cpp

```diff
@@ -169,97 +160,42 @@ void LocalPlanner::determineStrategy() {
       waypoint_type_ = direct;
     }
 
-    if (send_obstacles_fcu_) {
-      create2DObstacleRepresentation(true);
-    }
-  } else if (final_cloud_.points.size() > min_cloud_size_ && stop_in_front_ &&
-             reach_altitude_) {
-    obstacle_ = true;
-    ROS_INFO(
-        "\033[1;35m[OA] There is an Obstacle Ahead stop in front\n \033[0m");
-    stopInFrontObstacles();
-    waypoint_type_ = direct;
-
     if (send_obstacles_fcu_) {
       create2DObstacleRepresentation(true);
     }
```

Comment:
I'm guessing that 
```
    if (send_obstacles_fcu_) {
      create2DObstacleRepresentation(true);
    }
```
is the same as `create2DObstacleRepresentation(send_obstacles_fcu_)` ?

It seems that `create2DObstacleRepresentation(send_obstacles_fcu_)` is done both in the `if(!reach_altitude_)` and it's `else...`. Can we just do it once in the beginning?

Reply:
You would compute the histograms for no reasons during takeoff if you don't need to send the data to the Firmware